### PR TITLE
Feature/dark current noise

### PR DESCRIPTION
**Pull request by Kmudsam01**
* **Tickets addressed:** 513
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
This feature implements Dark Current Noise into the corruption pipeline. The dot product of the direction to the sun and the forward direction of the camera is used to determine the standard deviation of the Gaussian noise added to the image.  

## Verification
This code was verified by rotating the camera, and observing the printed sigma value. For the user to verify these changes, they must select the DarkCurrentNoise_Lvl by selecting File > Open Level. They must then begin the simulation by selecting the Play drop down menu (signified by the three dots next to the main play button) and selecting "Selected Viewport". The user can then press the "1" button and a number of times and observe the intended behavior: the Sigma value changing.

## Documentation
No documentation change is needed.

## Future work
incorporate other sources of heat, when calculating dark current noise.

### CODE REVIEW

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp
Comment (patkenneally) [inline Source/cielim/Private/RenderingFunctionsLibrary.cpp]: :large_orange_diamond: Because this line extends over 120 characters, you can drop each parameter onto a new line.

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp
Comment (patkenneally) [inline Source/cielim/Private/RenderingFunctionsLibrary.cpp]: 🔶 Because this line extends over 120 characters, you can drop each parameter onto a new line.

File: Source/cielim/Private/RenderingFunctionsLibrary.h
Comment (patkenneally) [inline Source/cielim/Private/RenderingFunctionsLibrary.h]: 🔶 Because this line extends over 120 characters, you can drop each parameter onto a new line.

**Review by patkenneally — APPROVED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp
Comment (Kmudsam01) [reply Source/cielim/Private/RenderingFunctionsLibrary.cpp]: Put each parameter onto a new line!

File: Source/cielim/Private/RenderingFunctionsLibrary.h
Comment (Kmudsam01) [reply Source/cielim/Private/RenderingFunctionsLibrary.h]: Put each parameter onto a new line!

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp
Comment (Kmudsam01) [reply Source/cielim/Private/RenderingFunctionsLibrary.cpp]: Put each parameter onto a new line!

**Pull request merged**
